compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -325,6 +338,11 @@ public void visit(BLangPackage pkgNode, AnalyzerData data) {
                 continue;
             }
 
+            // Analyze type definitions that get added while analyzing other nodes
+            if (pkgLevelNode.getKind() == NodeKind.TYPE_DEFINITION && i <= originalTopLevelNodeSize - 1) {
+                continue;
```

Comment:
Why are we special casing here? Why can't analyze all?

Reply:
Since we are analyzing the original type nodes before this for loop, there is no point in analyzing the same node twice.